src/traces/bar/plot.js

```diff
@@ -333,9 +487,20 @@ function appendBarText(gd, plotinfo, bar, cd, i, x0, x1, y0, y1, opts, makeOnCom
         }
     }
 
+    // Get width and height of bar at text position
+    var lx, ly, refPos;
+    if(isHorizontal) {
+        refPos = (insidetextanchor === 'middle') ? Math.abs(x1 - x0) / 2 : TEXTPAD;
+        lx = Math.abs(x1 - x0);
+        ly = lyFunc ? lyFunc(refPos) : Math.abs(y1 - y0);
+    } else {
+        refPos = (insidetextanchor === 'middle') ? Math.abs(y1 - y0) / 2 : TEXTPAD;
```

Comment:
This `refPos` for vertical bars is unused.

Reply:
@emilykl perhaps you wanted to pass it to `lxFunc` below?

Reply:
yes, you're right, looks like line 498 should read `lx = lxFunc ? lxFunc(refPos) : Math.abs(x1 - x0);`. Pushing that change now!